### PR TITLE
Add Kùzu remote backend examples

### PR DESCRIPTION
**Pull request by mewim**
This PR adds examples of Kùzu's remote backend integration with PyG.

### CODE REVIEW

File: examples/kuzu/Papers-100M/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Papers-100M example
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/README.md:1]: Can we rename the folder to `papers_100M`?

File: examples/kuzu/Papers-100M/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+This example shows how to use the remote backend feature of Kùzu to work with a large graph of papers and citations on a single machine. The data used in this example is `ogbn-papers100M` from the [Open Graph Benchmark](https://ogb.stanford.edu/). The dataset contains approximately 100 million nodes and 2.5 billion edges.
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/README.md:3]:
```suggestion
This example shows how to use the remote backend feature of [Kùzu](...) to work with a large graph of papers and citations on a single machine. The data used in this example is `ogbn-papers100M` from the [Open Graph Benchmark](https://ogb.stanford.edu/). The dataset contains approximately 100 million nodes and 2.5 billion edges.
```
Add a link to Kuzu.

File: examples/kuzu/Papers-100M/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+## Install the dependencies
+- numpy
+- pandas
+- kuzu
+- tqdm
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/README.md:9]:
```suggestion
pip install pandas kuzu
```

File: examples/kuzu/Papers-100M/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Prepare the data
+
+1. Download the dataset from [http://snap.stanford.edu/ogb/data/nodeproppred/papers100M-bin.zip](http://snap.stanford.edu/ogb/data/nodeproppred/papers100M-bin.zip). Put the zip file into this directory.
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/README.md:13]: Can't we use OGB directly here without manual download?

File: examples/kuzu/PubMed/PubMed.ipynb
Comment (rusty1s) [inline examples/kuzu/PubMed/PubMed.ipynb:1]: We do not include Notebooks into our code base. Can we either move this to a script as well, or move it to a Google Colab?

File: examples/kuzu/Papers-100M/train.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+# Get remote backend for torch geometric
+feature_store, graph_store = db.get_torch_geometric_remote_backend(
+    mp.cpu_count())
+
+# Plug the graph store and feature store into the NeighborLoader
+loader_kuzu = NeighborLoader(
+    data=(feature_store, graph_store), num_neighbors={
+        ('paper', 'cites', 'paper'): [30, 30]
+    }, batch_size=LOADER_BATCH_SIZE, input_nodes=('paper', input_nodes),
+    input_time=None, num_workers=4)
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/train.py:47]:
```suggestion
    input_time=None, num_workers=4, )
```
for nicer formatting.

File: examples/kuzu/Papers-100M/train.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+
+# Define the model for training. The model is ported from
+# https://github.com/mengyangniu/ogbn-papers100m-sage
+class MLP(nn.Module):
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/train.py:52]: You can use `torch_geometric.nn.models.MLP` for this.

File: examples/kuzu/Papers-100M/train.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+            self.bns.append(torch.nn.BatchNorm1d(n_hidden))
+        self.layers.append(
+            SAGEConv(n_hidden, n_hidden, aggr='mean', bias=False))
+        self.bns.append(torch.nn.BatchNorm1d(n_hidden))
+        self.mlp = MLP(in_feats + n_hidden * n_layers, 2 * n_classes,
+                       n_classes, num_layers=2, bn=True, end_up_with_fc=True,
+                       act='LeakyReLU')
+        self.dropout = nn.Dropout(dropout)
+        self.activation = activation
+        self.profile = locals()
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/train.py:97]: Can you clarify why we need this?

File: examples/kuzu/Papers-100M/train.py [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+criterion = torch.nn.CrossEntropyLoss()
+
+for epoch in range(NUM_EPOCHS):
+    i = 0
+    start_time = time.time()
+    for b in loader_kuzu:
+        x = b['paper']['x']
+        y = b['paper']['y']
+        edge_index = b['paper', 'cites', 'paper'].edge_index
+        y[torch.isnan(y)] = 171
```
Comment (rusty1s) [inline examples/kuzu/Papers-100M/train.py:132]: Can you clarify? All training labels should have a label, right?

File: examples/kuzu/Papers-100M/README.md (hunk 4 above)
Comment (mewim) [reply examples/kuzu/Papers-100M/README.md:13]: Does ogb provides a way of downloading the data without loading it into the RAM? When I called `dataset = PygNodePropPredDataset(name='ogbn-papers100M')`, it downloads the data and immediately starts to load and convert the data with a peak RAM usage of ~120 GB. This seem to defeat the purpose of using remote backend, as it would crash on systems without that much RAM.

File: examples/kuzu/PubMed/PubMed.ipynb
Comment (mewim) [reply examples/kuzu/PubMed/PubMed.ipynb:1]: We have this notebook hosted on Colab already. Could we link it directly in `README.md`?

Comment (rusty1s) [reply examples/kuzu/PubMed/PubMed.ipynb:1]: That would be perfect :)

File: examples/kuzu/Papers-100M/README.md (hunk 4 above)
Comment (rusty1s) [reply examples/kuzu/Papers-100M/README.md]: I see, I was somehow mistaken that OGB provides an interface to download without loading it into RAM :(

File: examples/kuzu/PubMed/PubMed.ipynb
Comment (mewim) [reply examples/kuzu/PubMed/PubMed.ipynb:1]: I have removed the notebook and added a Colab link in the README file.

File: examples/kuzu/Papers-100M/train.py (hunk 7 above)
Comment (mewim) [reply examples/kuzu/Papers-100M/train.py]: It was ported from the DGL example and I was not sure if/why this was needed. Since now I switched to use `torch_geometric.nn.models.MLP`, this is removed.

**Comment by mewim** [comment]
@rusty1s Thanks for your review. I think I have addressed the PR comments. Could you please take another look?

**Review by rusty1s — APPROVED** [review]
Thank you! Really great work :)
